_posts/2023-08-04-deploying-OIDC-webapp-on-openshift.adoc

```diff
@@ -0,0 +1,161 @@
+
+Now, click on `Roles` and create two roles, `user` and `admin`.
+
+Finally, create a user called `alice` whose `First Name` is Alice and `Last Name` is Smith and assign her the `user` and `admin` roles. Next, navigate to the `Credentials` tab and create a password for Alice. Toggle the `Temporary` switch off so you are not prompted to update the password after the first login. 
+
+== Building Container Image for WildFly
+
+In order to deploy our application on OpenShift, we need to first create a container image of the application using Docker or Podman. In this post we are using Docker, but you can also use Podman using the `podman` keyword, or using an alias.  
+
+Navigate to the `WildFly` project and create a `Dockerfile` inside the root directory of the project. We will be creating an image of the current version of WildFly using the `wildfly-runtime-jdk` image on quay. Quay is a repository for building and storing container images of your projects. 
+
+Head over to https://quay.io/repository/[quay.io] and create a repositories called `wildfly`. For more information on how to create one, visit the https://docs.quay.io/guides/create-repo.html#:~:text=via%20the%20UI-,To%20create%20a%20repository%20in%20the%20Quay.io%20UI%2C%20click,the%20'Create%20Repository'%20button.[quay documentation].
+
+Now we can create the container image for our WildFly project. Enter the following commands inside your Dockerfile: 
+```
+FROM quay.io/wildfly/wildfly-runtime-jdk11:latest
+COPY --chown=jboss:root dist/target/wildfly-29.0.0.Final-SNAPSHOT $JBOSS_HOME
+RUN chmod -R ug+rwX $JBOSS_HOME
+```
+Now run the following commands on your terminal from the WildFly directory to create an image and push it to your repository: 
+```
+docker build -t quay.io/<your-quay-username>/wildfly:wildfly-app .
+docker push quay.io/<your-quay-username>/wildfly:wildfly-app
+```
+The `-t` command indicates that we are specifying a tag, which in this case is `wildfly-app`. The tag is used to identify different images inside the same repository. Using this image, we can ensure that we are running your local version of WildFly, as opposed to the version of WildFly that is present on OpenShift. 
+
+== Deploying the Application 
+
+Now that we have our WildFly container image, we can use it to deploy our application on it. Navigate to the `elytron-examples/elytron-oidc-client-scope` directory and create a deployment file for your application using this command: 
+```
+mvn package -Pcloud
```

Comment:
Just to check, did the example you were following use "cloud" or "openshift"?

Reply:
Initially I did `OpenShift`, but changed the name of the layer to `cloud`

Reply:
Might be good to check if we actually need to specify the profile here and in the example pom. Since you are building the app manually and copying it manually to the wildfly deployments directory, it might not make sense to include it. (Normally, the `openshift` profile is included in an application pom when it will be built and deployed to a provisioned WildFly server with the specified layers using the WildFly Helm Chart and the wildfly-maven-plugin on OpenShift. In your case, you are creating the image manually.)